### PR TITLE
[EAGLE-3948] - Post model version upload

### PR DESCRIPTION
**Pull request by phatvo9**
### What
* Enable `create model version by uploading local file using new endpoint `PostModelVersionUpload`
* Other updates: move current model upload method to clarifai module

### How
* Implement `create_version_by_file` in Model class.
* Consume this feature in cli. Below is example usage.
```bash
$ clarifai upload model </path/to/your/project> --id model-id --user-app user-id/app-id
```

### CODE REVIEW

File: clarifai/client/model.py [hunk 1]
```diff
@@ -848,3 +849,319 @@ def _download_exported_model(
...
+
+    """
+    file_size = os.path.getsize(file_path)
+    # size >= 1GB
+    if file_size >= 1e9:
+      MIN_CHUNK = 1024 * 50_000  # 50MB
+    else:
+      MIN_CHUNK = 1024 * 10_000  # 10MB
+
+    MAX_SIZE_PER_STREAM = int(85_000_000)
```
[the remark this answers is not in the document]
Comment (phatvo9) [reply clarifai/client/model.py:949]: It has failed to upload > 100MB file in 1 stream, so I split it into smaller streams

File: clarifai/client/model.py [hunk 2]
```diff
@@ -848,3 +849,319 @@ def _download_exported_model(
...
+    Return:
+      Model: instance of Model with new created version
+
+    """
+    file_size = os.path.getsize(file_path)
+    # size >= 1GB
+    if file_size >= 1e9:
+      MIN_CHUNK = 1024 * 50_000  # 50MB
+    else:
+      MIN_CHUNK = 1024 * 10_000  # 10MB
```
[the remark this answers is not in the document]
Comment (phatvo9) [reply clarifai/client/model.py:947]: Just want to upload with bigger chunk size for large file

File: clarifai/client/model.py [hunk 3]
```diff
@@ -848,3 +849,319 @@ def _download_exported_model(
...
+    if not chunk_size:
+      chunk_size = MIN_CHUNK + 5
+
+    #self.logger.info(f"Chunk {chunk_size/1e6}MB, {file_size/chunk_size} steps")
+    #self.logger.info(f" Max bytes per stream {MAX_SIZE_PER_STREAM}")
+
+    if chunk_size < MIN_CHUNK:
+      raise ValueError("Chunk size must be at least 5MB")
+
+    cache_dir = os.path.join(file_path, '..', '.cache')
```
[the remark this answers is not in the document]
Comment (phtvo) [reply clarifai/client/model.py:964]: it should locate same place with built cache

File: clarifai/client/model.py [hunk 4]
```diff
@@ -848,3 +849,319 @@ def _download_exported_model(
...
+              user_app_id=self.user_app_id,
+              model_id=self.id,
+              total_size=file_size,
+              model_version=resources_pb2.ModelVersion(
+                  id=model_version,
+                  pretrained_model_config=pretrained_proto,
+                  description=description,
+                  output_info=resources_pb2.OutputInfo(params_specs=inference_param_proto))),
+          content_part=resources_pb2.UploadContentPart(
+              data=chunk, part_number=part_id, range_start=range_start))
```
[the remark this answers is not in the document]
Comment (phtvo) [reply clarifai/client/model.py:1005]: thanks, I removed `upload_config` in _uploading

File: clarifai/client/model.py [hunk 5]
```diff
@@ -848,3 +853,302 @@ def _download_exported_model(
...
+        try:
+          cache_info = json.load(fp)
+          if isinstance(cache_info, dict):
+            part_id = cache_info.get("part_id", part_id)
+            chunk_size = cache_info.get("chunk_size", chunk_size)
+            range_start = cache_info.get("range_start", range_start)
+            model_version = cache_info.get("model_version", model_version)
+            last_percent = cache_info.get("last_percent", last_percent)
+        except Exception:
+          pass
```
Comment (ackizilkale) [inline clarifai/client/model.py:974]: do we want to do error handling here?

**Review by ackizilkale — APPROVED** [review]
looks good

**Review by wemoveon2 — APPROVED** [review]

**Pull request merged**